packages/safe-ds-vscode/src/extension/mainClient.ts

```diff
@@ -128,29 +128,70 @@ const acceptRunRequests = function (context: vscode.ExtensionContext) {
-            const pipelineId = crypto.randomUUID();
-            printOutputMessage(`Launching Pipeline (${pipelineId}): ${pipelinePath}`);
-            executePipeline(services, pipelinePath.fsPath, pipelineId);
-        }),
+        vscode.commands.registerCommand(
+            'extension.safe-ds.runPipelineFile',
+            async (filePath: vscode.Uri | undefined) => {
+                let pipelinePath = filePath;
+                // Allow execution via command menu
+                if (!pipelinePath && vscode.window.activeTextEditor) {
+                    pipelinePath = vscode.window.activeTextEditor.document.uri;
+                }
+                if (
+                    pipelinePath &&
+                    !services.LanguageMetaData.fileExtensions.some((extension: string) =>
+                        pipelinePath!.fsPath.endsWith(extension),
+                    )
+                ) {
+                    vscode.window.showErrorMessage(`Could not run ${pipelinePath!.fsPath} as it is not a Safe-DS file`);
+                    return;
+                }
+                if (!pipelinePath) {
+                    vscode.window.showErrorMessage(
+                        'Could not run Safe-DS Pipeline, as no pipeline is currently selected.',
+                    );
+                    return;
+                }
+                // Refresh workspace
+                // Do not delete builtins
+                services.shared.workspace.LangiumDocuments.all
+                    .filter(
+                        (document) =>
+                            !(
+                                ast.isSdsModule(document.parseResult.value) &&
+                                (<ast.SdsModule>document.parseResult.value).name === 'safeds.lang'
+                            ),
+                    )
+                    .forEach((oldDocument) => {
+                        services.shared.workspace.LangiumDocuments.deleteDocument(oldDocument.uri);
+                    });
+                const workspaceSdsFiles = await vscode.workspace.findFiles('**/*.{sdspipe,sdsstub,sdstest}');
+                // Load all documents
+                const unvalidatedSdsDocuments = workspaceSdsFiles.map((newDocumentUri) =>
+                    services.shared.workspace.LangiumDocuments.getOrCreateDocument(newDocumentUri),
+                );
+                // Validate them
+                const validationErrorMessage = await validateDocuments(services, unvalidatedSdsDocuments);
+                if (validationErrorMessage) {
+                    vscode.window.showErrorMessage(validationErrorMessage);
+                    return;
+                }
+                // Run it
+                const pipelineId = crypto.randomUUID();
+                printOutputMessage(`Launching Pipeline (${pipelineId}): ${pipelinePath}`);
+                let mainDocument;
+                if (!services.shared.workspace.LangiumDocuments.hasDocument(URI.parse(pipelinePath.toString()))) {
```

Comment:
```suggestion
                if (!services.shared.workspace.LangiumDocuments.hasDocument(pipelinePath)) {
```

Doesn't this work too?